.github/workflows/main.yml

```diff
@@ -0,0 +1,71 @@
+
+name: buildx
+
+on:
+  push:
+    tags:
+      - v*
+
+jobs:
+  buildx:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Prepare
+        id: prepare
+        run: |
+          DOCKER_IMAGE=${GITHUB_REPOSITORY}
+          DOCKER_PLATFORMS=linux/amd64,linux/arm/v7,linux/arm64
+          VERSION=edge
+
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            VERSION=${GITHUB_REF#refs/tags/}
+          fi
+
+          TAGS="--tag ${DOCKER_IMAGE}:${VERSION}"
+          if [[ $VERSION =~ ^v[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+            TAGS="$TAGS --tag ${DOCKER_IMAGE}:latest"
+          fi
+
+          echo ::set-output name=docker_image::${DOCKER_IMAGE}
+          echo ::set-output name=version::${VERSION}
+          echo ::set-output name=buildx_args::--platform ${DOCKER_PLATFORMS} \
+            --build-arg VERSION=${VERSION} \
+            --build-arg BUILD_DATE=$(date -u +'%Y-%m-%dT%H:%M:%SZ') \
+            --build-arg VCS_REF=${GITHUB_SHA::8} \
+            ${TAGS} --file ./Dockerfile .
+      -
+        name: Set up Docker Buildx
+        uses: crazy-max/ghaction-docker-buildx@v3
+      -
+        name: Docker Buildx (build)
+        run: |
+          echo '${{ steps.prepare.outputs.buildx_args }}'|awk '{print $4}'|awk -F= '{print $2}' |tee RELEASE 
```

Comment:
What does this RELEASE stuff do?

Reply:
Do [this](https://github.com/syncthing/syncthing/blob/55147f5901ca8be3d6c06762bbec2f4fd6e42b82/build.go#L854), content is tag name. eg: `v1.8.0-rc.2`
If not ,[getGitVersion()](https://github.com/syncthing/syncthing/blob/55147f5901ca8be3d6c06762bbec2f4fd6e42b82/build.go#L862) will be get commit sha for version, it's not allowed.
I think this is a bug(even if you can use the RELEASE file to skip), because it will cause developers to be unable to checkout the tag to compile the docker image.

Reply:
If we need a version name somewhere, the canonical would be to get it from `go run build.go version` so that it's the same as the binary inside the image.

Reply:
In github action checkout command is:
```bash
git checkout --progress --force refs/tags/v1.8.0-rc.5
```
but `build.go` get a commit sha name(eg: 5c01c3d8), I had to use the RELEASE file can be work.